zrml/rikiddo/src/utils.rs

```diff
@@ -0,0 +1,22 @@
+//! This module contains utility functions for creating commonly used
+//! fixed point type constants.
+
+use substrate_fixed::traits::Fixed;
+
+/// Create a fixed point number that represents 0 (zero).
+pub fn fixed_zero<FixedType: Fixed>() -> Result<FixedType, &'static str> {
+    if let Some(res) = FixedType::checked_from_num(0) {
+        Ok(res)
+    } else {
+        Err("Unexpectedly failed to convert zero to fixed point type")
+    }
+}
+
+/// Return the maximum value of FixedType as u128.
+pub fn max_value_u128<FixedType: Fixed>() -> Result<u128, &'static str> {
+    if let Some(res) = FixedType::max_value().int().checked_to_num() {
+        Ok(res)
+    } else {
+        Err("Unexpectedly failed to convert max_value of fixed point type to u128")
+    }
+}
```

Comment:
Since you created a new module for utility functions, I think we should also move the following function into that module: https://github.com/zeitgeistpm/zeitgeist/blob/cd55ba8a6af316993b288b319436e4b55cddd585/zrml/rikiddo/src/types.rs#L127-L172

Reply:
Not a 100% sure about the following, but I tend towards yes: https://github.com/zeitgeistpm/zeitgeist/blob/cd55ba8a6af316993b288b319436e4b55cddd585/zrml/rikiddo/src/types.rs#L174-L299

Reply:
Agreed. Thinking about this in terms of encapsulation, I would consider these functions to be implementation details of `types.rs`, so putting them in the utilities seems correct.